code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm

```diff
@@ -0,0 +1,201 @@
+
+/mob/living/simple_animal/hostile/megafauna/wendigo/Initialize()
+	. = ..()
+	starting = get_turf(src)
+
+/mob/living/simple_animal/hostile/megafauna/wendigo/OpenFire()
+	SetRecoveryTime(0, 100)
+	if(health <= maxHealth*0.5)
+		stomp_range = 2
+		speed = 6
+		move_to_delay = 6
+	else
+		stomp_range = initial(stomp_range)
+		speed = initial(speed)
+		move_to_delay = initial(move_to_delay)
+
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				heavy_stomp()
+			if(2)
+				teleport()
+			if(3)
+				disorienting_scream()
+		return
+
+	chosen_attack = rand(1, 3)
+	switch(chosen_attack)
+		if(1)
+			heavy_stomp()
+		if(2)
+			teleport()
+		if(3)
+			disorienting_scream()
+
+/mob/living/simple_animal/hostile/megafauna/wendigo/Move(atom/newloc, direct)
+	if(!can_move)
+		return
+	stored_move_dirs |= direct
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/wendigo/Moved(atom/oldloc, direct)
+	. = ..()
+	stored_move_dirs &= ~direct
+	if(!stored_move_dirs)
+		INVOKE_ASYNC(src, .proc/ground_slam, stomp_range, 1)
+
+/// Slams the ground around the wendigo throwing back enemies caught nearby
+/mob/living/simple_animal/hostile/megafauna/wendigo/proc/ground_slam(range, delay)
+	var/turf/orgin = get_turf(src)
+	var/list/all_turfs = RANGE_TURFS(range, orgin)
+	for(var/i = 0 to range)
+		for(var/turf/T in all_turfs)
+			if(get_dist(orgin, T) > i)
+				continue
+			playsound(T,'sound/effects/bamf.ogg', 600, TRUE, 10)
+			new /obj/effect/temp_visual/small_smoke/halfsecond(T)
+			for(var/mob/living/L in T)
+				if(L == src || L.throwing)
+					continue
+				to_chat(L, "<span class='userdanger'>[src]'s ground slam shockwave sends you flying!</span>")
+				var/turf/thrownat = get_ranged_target_turf_direct(src, L, 8, rand(-10, 10))
+				L.throw_at(thrownat, 8, 2, src, TRUE)		//, force = MOVE_FORCE_OVERPOWERING, gentle = TRUE)
+				L.apply_damage(20, BRUTE)
+				shake_camera(L, 2, 1)
+			all_turfs -= T
+		sleep(delay)
+
+/// Larger but slower ground stomp
+/mob/living/simple_animal/hostile/megafauna/wendigo/proc/heavy_stomp()
+	can_move = FALSE
+	ground_slam(5, 2)
+	SetRecoveryTime(0, 0)
+	can_move = TRUE
+
+/// Teleports to a location 4 turfs away from the enemy in view
+/mob/living/simple_animal/hostile/megafauna/wendigo/proc/teleport()
+	var/list/possible_ends = list()
+	for(var/turf/T in view(4, target.loc) - view(3, target.loc))
+		if(isclosedturf(T))
+			continue
+		possible_ends |= T
+	var/turf/end = pick(possible_ends)
+	do_teleport(src, end, 0,  channel=TELEPORT_CHANNEL_BLUESPACE, forced = TRUE)
+	SetRecoveryTime(20, 0)
+
+/// Shakes all nearby enemies screens and animates the wendigo shaking up and down
+/mob/living/simple_animal/hostile/megafauna/wendigo/proc/disorienting_scream()
+	can_move = FALSE
+	playsound(src, 'sound/magic/demon_dies.ogg', 600, FALSE, 10)
+	animate(src, pixel_z = rand(5, 15), time = 1, loop = 6)
+	animate(pixel_z = 0, time = 1)
+	for(var/mob/living/L in get_hearers_in_view(7, src) - src)
+		shake_camera(L, 120, 2)
```

Comment:
screenshake is so high that it crashes games and also makes it impossible to hit the megafauna in melee even if your game didn't go kaput

Reply:
you should also remove the screenshake from the movement ability because hoooooooooly fuck hahaaaaaaaaaaaaaaaaaaaaaaaaaaa